docs/data/base/components/input/OTPInput.tsx

```diff
@@ -0,0 +1,254 @@
+          return otpArray;
+        });
+
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    currentIndex: number,
+  ) => {
+    const currentValue = event.target.value;
+    let indexToEnter = 0;
+
+    while (indexToEnter <= currentIndex) {
+      if (value[indexToEnter] && indexToEnter < currentIndex) {
+        indexToEnter += 1;
+      } else {
+        break;
+      }
+    }
+    onChange((prev) => {
+      const otpArray = [...prev];
+      const lastValue = currentValue[currentValue.length - 1];
+      otpArray[indexToEnter] = lastValue;
+      return otpArray;
+    });
+    if (currentValue !== '') {
+      if (currentIndex < inputCount - 1) {
+        focusInput(currentIndex + 1);
+      }
+    }
+  };
+
+  const handleClick = (
+    event: React.MouseEvent<HTMLInputElement, MouseEvent>,
+    currentIndex: number,
+  ) => {
+    selectInput(currentIndex);
+  };
+
+  const handlePaste = (
+    event: React.ClipboardEvent<HTMLInputElement>,
+    currentIndex: number,
+  ) => {
+    event.preventDefault();
+    const clipboardData = event.clipboardData;
+
+    // Check if there is text data in the clipboard
+    if (clipboardData.types.includes('text/plain')) {
+      let pastedText = clipboardData.getData('text/plain');
+      pastedText = pastedText.substring(0, inputCount);
+      let indexToEnter = 0;
+
+      while (indexToEnter <= currentIndex) {
+        if (value[indexToEnter] && indexToEnter < currentIndex) {
+          indexToEnter += 1;
+        } else {
+          break;
+        }
+      }
+
+      const otpArray = [...value];
+
+      for (let i = indexToEnter; i < inputCount; i += 1) {
+        const lastValue = pastedText[i - indexToEnter] ?? '';
+        otpArray[i] = lastValue;
+      }
+
+      onChange(otpArray);
+    }
+  };
+
+  return (
+    <Box sx={{ display: 'flex', gap: 1, alignItems: 'center' }}>
+      {new Array(inputCount).fill(null).map((_, index) => (
+        <React.Fragment key={index}>
+          <BaseInput
+            slots={{
+              input: InputElement,
+            }}
+            aria-label={`Digit ${index + 1} of OTP`}
+            slotProps={{
+              input: {
+                ref: (ele) => {
+                  inputRefs.current[index] = ele!;
+                },
+                onKeyDown: (event) => handleKeyDown(event, index),
+                onChange: (event) => handleChange(event, index),
+                onClick: (event) => handleClick(event, index),
+                onPaste: (event) => handlePaste(event, index),
+                value: value[index],
+              },
+            }}
+          />
+          {index === inputCount - 1 ? null : seperator}
+        </React.Fragment>
+      ))}
+    </Box>
+  );
+}
+
+export default function OTPInput() {
+  const inputCount = 5;
+  const [otp, setOtp] = React.useState<string[]>(new Array(inputCount).fill(''));
```

Comment:
Having just a string here instead of an array would be more convenient. The value would usually be sent to server as a single string.